pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -183,8 +200,18 @@ func (rc *reconciler) reconcile() {
 			}
 			// Check whether timeout has reached the maximum waiting time
 			timeout := elapsedTime > rc.maxWaitForUnmountDuration
+
+			hasOutOfServiceTaint := false
+			if utilfeature.DefaultFeatureGate.Enabled(features.NodeOutOfServiceVolumeDetach) {
```

Comment:
nit: can you move feature gate check inside `hasOutOfServiceTaint` function, so as it always returns false if feature gate is not enabled?